src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,183 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# Entries (p5 Template)
+
+[<img
+  src="../../assets/img/Entries/1729.png"
+  width=700
+  style="margin-left: 1em; margin-right: 0.5em"
+/>](../../assets/img/Entries/1729.png)
+
+This visualizer allows you to step back and forth through a sequence, showing
+each entry as a written number.
+
+_This visualizer is meant to be used as a template for new visualizers based on
+the p5.js library. It includes explanatory comments and minimal examples of
+required and commonly used features._
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    // === visualizer name ===
+    // appears in the visualizer list and bundle card titles
+
+    static visualizationName = 'Entries (p5 Template)'
+
+    // === parameters ===
+    // top-level properties that the user can choose while creating the
+    // visualizer bundle
+
+    stepSize = 1 // the default value of this parameter is zero
+
+    /** md
+## Parameters
+    **/
+    params = {
+        /** md
+- **Step size:** How far to step when the user presses an arrow key.
+         **/
+        stepSize: {
+            value: this.stepSize,
+            forceType: 'integer',
+            displayName: 'Step size',
+            required: true,
+        },
+    }
+
+    // === internal properties ===
+    // top-level properties that are set and updated while the visualizer is
+    // running, beyond the user's direct control
+
+    // navigation state. this property will be initialized during setup, but
+    // TypeScript can't infer that, so we have to give a placeholder value
+    index = 0
+
+    // palette colors. these will also be initialized during setup, which is our
+    // first chance to create colors. for now, we use `undefined` as a
+    // placeholder value
+    bgColor: p5.Color | undefined = undefined
+    textColor: p5.Color | undefined = undefined
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the step size is positive
+        if (this.params.stepSize.value <= 0) {
+            status.isValid = false
+            status.errors.push('Step size must be positive')
+        }
+
+        return status
+    }
+
+    setup() {
+        // the base setup function includes the `createCanvas()` call that must
+        // appear in every p5 setup function
+        //
+        //   https://p5js.org/reference/#/p5/createCanvas
+        //
+        super.setup()
+
+        const sketch = this.sketch
+
+        // initialize palette colors, chosen from the Numberscope site palette
+        this.bgColor = sketch.color(206, 212, 218)
+        this.textColor = sketch.color(128, 159, 255)
+        const outlineColor = sketch.color(51, 51, 255)
+
+        // set the stroke color and text alignment, which won't change from
+        // frame to frame. each setting method returns a reference to the
+        // sketch, so you can chain the methods as shown here
```

Comment:
Capitalization, as these are complete sentences. I don't mind all lower case and no period in comments that are sentence fragments.

Reply:
All good on capitalization/punctuation now.